tiledb/sm/c_api/tiledb.h

```diff
@@ -4620,6 +4630,8 @@ TILEDB_EXPORT int32_t tiledb_array_open_with_key(
     uint32_t key_length);
 
 /**
+ * This is a deprecated API.
```

Comment:
Need to add `TILEDB_DEPRECATED_EXPORT`.

Reply:
Note: Use of `TILEDB_DEPRECATED_EXPORT` in `tiledb/sm/c_api/array.h` is causing the Windows CI jobs to fail. This macro has been removed for the time being, and the APIs are "deprecated" with block comments.